Formula/o/onctl.rb

```diff
@@ -0,0 +1,24 @@
+class Onctl < Formula
+  desc "Command-line tool for managing cloud resources with onctl"
+  homepage "https://github.com/cdalar/onctl"
+  url "https://github.com/cdalar/onctl.git",
+    tag:      "v0.1.19",
+    revision: "6323244fda55d7edff4c5f2718f1c7eaa9cd772d"
+  license "AGPL-3.0-or-later"
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/cdalar/onctl/cmd.Version=#{version}
+      -X github.com/cdalar/onctl/cmd.BuildTime=#{Time.now.utc.strftime("%Y-%m-%d %H:%M:%S")}
+      -X github.com/cdalar/onctl/cmd.GoVersion=#{Formula["go"].version}
+    ]
+    system "go", "mod", "tidy"
+    system "go", "build", *std_go_args(ldflags:), "-o", "onctl"
+  end
+
+  test do
+    system "bin/onctl", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.